### PR TITLE
-json option on jobs status command

### PR DESCRIPTION
**Pull request by geovannyAvelar**
https://github.com/hashicorp/nomad/issues/16566

This modification dumps the entire `job` struct in JSON format. I don't know if just dumps the entire struct is the better approach, if not, i can expose specific fields.

### CODE REVIEW

**Review by tgross — COMMENTED** [review]
Hi @geovannyAvelar! I know this is still in draft but just a heads up that the approach you've got here is essentially what the existing `nomad job inspect` does. The goal described in https://github.com/hashicorp/nomad/issues/16566 is to have the whole command's output in some JSON-structured way, which includes a lot of ancillary information like the allocations, deployments, etc.

**Comment by geovannyAvelar** [comment]
Hello @tgross! I made some changes. Now when `-json` option is included in `nomad job status` command, we output same data of `nomad job status`  command (evals, allocations, latest deployment and summary). Please let me know if I need to change or include something. Do I need to include something different when handling periodic or parameterized jobs?

File: command/job_status.go [hunk 1]
```diff
@@ -666,6 +718,51 @@ func (c *JobStatusCommand) outputFailedPlacements(failedEval *api.Evaluation) {
...
+	for i, id := range jobIds {
+		job, _, err := client.Jobs().Info(id, &api.QueryOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("Error querying job info: %s", err)
+		}
+
+		var q *api.QueryOptions
+		if job.Namespace != nil {
+			q = &api.QueryOptions{Namespace: *job.Namespace}
+		}
```
Comment (tgross) [inline command/job_status.go:733]: I did some testing of this branch with jobs in namespaces and discovered that we're not passing in the correct namespace for each job ID we're querying, and instead end up picking up the namespace set earlier. What that results in is wildcard namespaces being broken with the `-json` flag:

```
$ nomad job status -namespace '*'
ID      Namespace  Type     Priority  Status          Submit Date
httpd   default    service  50        dead (stopped)  2024-02-22T12:53:17-05:00
httpd   prod       service  50        running         2024-02-22T12:53:35-05:00
system  default    system   50        dead (stopped)  2024-02-22T12:53:20-05:00

$ nomad job status -json -namespace '*'
Error querying job info: Unexpected response code: 404 (job not found)
```

So I think we need to pass in namespace/ID pairs to this function.

**Review by tgross — CHANGES_REQUESTED** [review]

File: command/job_status.go [hunk 2]
```diff
@@ -26,6 +26,20 @@ type JobStatusCommand struct {
...
+	Summary          *api.JobSummary
+	Allocations      []*api.AllocationListStub
+	LatestDeployment *api.Deployment
+	Evaluations      []*api.Evaluation
+}
+
+type jobIDNamespacePair struct {
+	ID        string
+	Namespace string
 }
```
Comment (tgross) [inline command/job_status.go:43]: Upon seeing this struct and thinking "man, we must have this already, right?" I remembered we have `structs.NamespacedID` for just this purpose. Let's use that here.

**Review by tgross — COMMENTED** [review]
This is looking great @geovannyAvelar. To get this merged:
* Run `make cl` to add a changelog entry
* Add documentation for the new command flag in `website/content/docs/commands/job/status.mdx`
* Mark the PR as ready-for-review

**Review by tgross — APPROVED** [review]
LGTM! 

I've made a minor adjustment to the changelog entry. Once CI is green I'll get this merged for the next 1.7.x release. Thanks @geovannyAvelar!

**Comment by tgross** [comment]
Bah, sorry @geovannyAvelar, it looks like I led you astray with the `structs` package, as we have a linter than prevents that import (see https://github.com/hashicorp/nomad/pull/13938, which I approved :facepalm: ). I've fixed that up for you.

**Pull request merged**